sdks/python/apache_beam/runners/interactive/utils_test.py

```diff
@@ -0,0 +1,138 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+import unittest
+
+import pandas as pd
+
+from apache_beam.runners.interactive import utils
+from apache_beam.typehints.typehints import Any
+from apache_beam.typehints.typehints import Dict
+from apache_beam.typehints.typehints import List
+from apache_beam.typehints.typehints import Tuple
+from apache_beam.utils.windowed_value import WindowedValue
+
+
+class ParseToDataframeTest(unittest.TestCase):
+  def test_parse_1(self):
+    el = (1, 'a')
+    element_type = Tuple[int, str]
+
+    columns, elements = utils.parse_row(el, element_type)
+    self.assertEqual(columns, ['el[0]', 'el[1]'])
+    self.assertEqual(elements, [1, 'a'])
+
+  def test_parse_2(self):
+    el = ((1, 2.0, 'a'), 'b')
+    element_type = Tuple[Tuple[int, float, str], str]
+
+    columns, elements = utils.parse_row(el, element_type)
+    self.assertEqual(columns, ['el[0][0]', 'el[0][1]', 'el[0][2]', 'el[1]'])
+    self.assertEqual(elements, [1, 2.0, 'a', 'b'])
+
+  def test_parse_3(self):
+    el = [1, 2, 3]
+    element_type = List[int]
+
+    columns, elements = utils.parse_row(el, element_type)
+    self.assertEqual(columns, ['el'])
+    self.assertEqual(elements, [[1, 2, 3]])
+
+  def test_parse_4(self):
+    el = ('k', [1, 2, 3])
+    element_type = Tuple[str, List[int]]
+
+    columns, elements = utils.parse_row(el, element_type)
+    self.assertEqual(columns, ['el[0]', 'el[1]'])
+    self.assertEqual(elements, ['k', [1, 2, 3]])
+
+  def test_parse_5(self):
+    el = (([1, 2, 3], {'b': 1, 'c': 2}), 'a')
+    element_type = Tuple[Tuple[List[int], Dict[str, int]], str]
+
+    columns, elements = utils.parse_row(el, element_type)
+    self.assertEqual(columns, ['el[0][0]', 'el[0][1]', 'el[1]'])
+    self.assertEqual(elements, [[1, 2, 3], {'b': 1, 'c': 2}, 'a'])
+
+  def test_parse_5(self):
```

Comment:
Can the tests be assigned meaningful names? Also `test_parse_5` occurred twice.

Reply:
Done